vsphere/resource_vsphere_tag_category.go

```diff
@@ -0,0 +1,251 @@
+	vSphereTagCategoryAssociableTypeHostSystem                     = "HostSystem"
+	vSphereTagCategoryAssociableTypeContentLibrary                 = "com.vmware.content.Library"
+	vSphereTagCategoryAssociableTypeContentLibraryItem             = "com.vmware.content.library.Item"
+	vSphereTagCategoryAssociableTypeHostNetwork                    = "HostNetwork"
+	vSphereTagCategoryAssociableTypeNetwork                        = "Network"
+	vSphereTagCategoryAssociableTypeOpaqueNetwork                  = "OpaqueNetwork"
+	vSphereTagCategoryAssociableTypeResourcePool                   = "ResourcePool"
+	vSphereTagCategoryAssociableTypeVirtualApp                     = "VirtualApp"
+	vSphereTagCategoryAssociableTypeVirtualMachine                 = "VirtualMachine"
+
+	vSphereTagCategoryAssociableTypeAll = "All"
+)
+
+// The following groups are type groups that are associated with the same type
+// selection in the vSphere Client tag category UI.
+var (
+	// vSphereTagCategoryAssociableTypesForDistributedVirtualSwitch represents
+	// types for virtual switches.
+	vSphereTagCategoryAssociableTypesForDistributedVirtualSwitch = []string{
+		vSphereTagCategoryAssociableTypeDistributedVirtualSwitch,
+		vSphereTagCategoryAssociableTypeVmwareDistributedVirtualSwitch,
+	}
+
+	// vSphereTagCategoryAssociableTypesForNetwork represents the types for
+	// networks.
+	vSphereTagCategoryAssociableTypesForNetwork = []string{
+		vSphereTagCategoryAssociableTypeHostNetwork,
+		vSphereTagCategoryAssociableTypeNetwork,
+		vSphereTagCategoryAssociableTypeOpaqueNetwork,
+	}
+)
+
+func resourceVSphereTagCategory() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVSphereTagCategoryCreate,
+		Read:   resourceVSphereTagCategoryRead,
+		Update: resourceVSphereTagCategoryUpdate,
+		Delete: resourceVSphereTagCategoryDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVSphereTagCategoryImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The display name of the category.",
+				Required:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "The description of the category.",
+				Optional:    true,
+			},
+			"cardinality": {
+				Type:        schema.TypeString,
+				Description: "The associated cardinality of the category. Can be one of SINGLE (object can only be assigned one tag in this category) or MULTIPLE (object can be assigned multiple tags in this category).",
+				ForceNew:    true,
+				Required:    true,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						vSphereTagCategoryCardinalitySingle,
+						vSphereTagCategoryCardinalityMultiple,
+					},
+					false,
+				),
+			},
+			"associable_types": {
+				Type:        schema.TypeSet,
+				Description: "Object types to which this category's tags can be attached.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    true,
+			},
+			"used_by": {
```

Comment:
I feel like this is named incorrectly?

Reply:
Yeah, I almost feel like removing this field altogether. I will do a bit more research on it. [API entry where it's referenced](https://vdc-repo.vmware.com/vmwb-repository/dcr-public/1cd28284-3b72-4885-9e31-d1c6d9e26686/71ef7304-a6c9-43b3-a3cd-868b2c236c81/doc/operations/com/vmware/cis/tagging/category.get-operation.html).